apps/explorer/src/ui/CoinBalance.tsx

```diff
@@ -1,19 +1,61 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { cva, type VariantProps } from 'class-variance-authority';
+
+import {
+    useFormatCoin,
+    CoinFormat,
+    formatBalance,
+} from '~/hooks/useFormatCoin';
 import { Heading } from '~/ui/Heading';
-import { Text } from '~/ui/Text';
+
+const coinBalanceStyles = cva('', {
+    variants: {
+        format: {
+            large: 'heading2',
+            medium: 'heading6',
+        },
+    },
+    defaultVariants: {
+        format: 'medium',
+    },
+});
+export interface CoinBalanceProps
+    extends VariantProps<typeof coinBalanceStyles> {
+    amount: number | string | bigint;
+    symbol?: string | null;
+}
+
+const DECIMALS = 0;
 
 // Passing amount as a string or number for optional number suffix
-export type CoinBalanceProps = { amount: number | string; symbol?: string };
+export function CoinBalance({ amount, symbol, format }: CoinBalanceProps) {
+    const [formattedAmount, suffix] = useFormatCoin(
```

Comment:
If we're doing data fetching here, we should split this into two components: one for just the presentation, and one that wraps it and uses the formatting hook.

Reply:
Ok. I will Create a new PR to separate the two.